src/scene/export.ts

```diff
@@ -10,6 +10,8 @@ import { AppState } from "../types";
 import { t } from "../i18n";
 import { DEFAULT_FONT_FAMILY } from "../appState";
 
+import jsPDF from "jspdf";
```

Comment:
jspdf is huge

```js
290.3
kB
MINIFIED
87.6
kB
MINIFIED + GZIPPED
```

Since you're only using it to export a static image, could you instead inline the actual pdf it generates?

```js
%PDF-1.3
%ºß¬à
3 0 obj
<</Type /Page
/Parent 1 0 R
/Resources 2 0 R
/MediaBox [0 0 404.92 399.73]
/Contents 4 0 R
>>
endobj
4 0 obj
<<
/Length 53
>>
stream
0.20 w
0 G
q
399.73 0 0 404.92 0.00 -5.19 cm
/I0 Do
Q
endstream
endobj
1 0 obj
<</Type /Pages
/Kids [3 0 R ]
/Count 1
>>
endobj
5 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
6 0 obj
<<
/Type /Font
/BaseFont /Helvetica-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
7 0 obj
<<
/Type /Font
/BaseFont /Helvetica-Oblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
8 0 obj
<<
/Type /Font
/BaseFont /Helvetica-BoldOblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
9 0 obj
<<
/Type /Font
/BaseFont /Courier
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Courier-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
11 0 obj
<<
/Type /Font
/BaseFont /Courier-Oblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
12 0 obj
<<
/Type /Font
/BaseFont /Courier-BoldOblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
13 0 obj
<<
/Type /Font
/BaseFont /Times-Roman
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
14 0 obj
<<
/Type /Font
/BaseFont /Times-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
15 0 obj
<<
/Type /Font
/BaseFont /Times-Italic
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
16 0 obj
<<
/Type /Font
/BaseFont /Times-BoldItalic
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
17 0 obj
<<
/Type /Font
/BaseFont /ZapfDingbats
/Subtype /Type1
/FirstChar 32
/LastChar 255
>>
endobj
18 0 obj
<<
/Type /Font
/BaseFont /Symbol
/Subtype /Type1
/FirstChar 32
/LastChar 255
>>
endobj
19 0 obj
<<
/Type /XObject
/Subtype /Image
/Width 799
/Height 809
/ColorSpace /DeviceRGB
/BitsPerComponent 8
/Length 51257
/Filter /DCTDecode
>>
stream
                    ... binary file goes here ...
endstream
endobj
2 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/Font <<
/F1 5 0 R
/F2 6 0 R
/F3 7 0 R
/F4 8 0 R
/F5 9 0 R
/F6 10 0 R
/F7 11 0 R
/F8 12 0 R
/F9 13 0 R
/F10 14 0 R
/F11 15 0 R
/F12 16 0 R
/F13 17 0 R
/F14 18 0 R
>>
/XObject <<
/I0 19 0 R
>>
>>
endobj
20 0 obj
<<
/Producer (jsPDF 1.5.3)
/CreationDate (D:20200622092449-07'00')
>>
endobj
21 0 obj
<<
/Type /Catalog
/Pages 1 0 R
/OpenAction [3 0 R /FitH null]
/PageLayout /OneColumn
>>
endobj
xref
0 22
0000000000 65535 f 
0000000227 00000 n 
0000053471 00000 n 
0000000015 00000 n 
0000000124 00000 n 
0000000284 00000 n 
0000000409 00000 n 
0000000539 00000 n 
0000000672 00000 n 
0000000809 00000 n 
0000000932 00000 n 
0000001061 00000 n 
0000001193 00000 n 
0000001329 00000 n 
0000001457 00000 n 
0000001584 00000 n 
0000001713 00000 n 
0000001846 00000 n 
0000001948 00000 n 
0000002044 00000 n 
0000053730 00000 n 
0000053816 00000 n 
trailer
<<
/Size 22
/Root 21 0 R
/Info 20 0 R
/ID [ <142D0971D336356A9C4483D569765BFF> <142D0971D336356A9C4483D569765BFF> ]
>>
startxref
53920
%%EOF
```

Reply:
Better to just lazyload the library. Though, not 100% sure if the FS API will like the call not coming from the event handler.

Reply:
No. It's way better to inline the generated pdf that to lazyload a 100kb library.

Reply:
Right, but how do you inline the PDF renderer output? From what you posted above, nothing springs to mind. But maybe you're no longer seeing just the matrix source code...

Reply:
I quoted the entire pdf file. I would just do

const pdffile = "giant prelude" + binary of the image + "giant postlude";

Reply:
Oh, if it's just pasting raw PNG binary data, then for sure. Can we even parse & encode binary data in JS?